lib/libzpool/kernel.c

```diff
@@ -516,10 +339,9 @@ cv_timedwait(kcondvar_t *cv, kmutex_t *mp, clock_t abstime)
 		ts.tv_nsec -= NANOSEC;
 	}
 
-	ASSERT3P(mutex_owner(mp), ==, curthread);
```

Comment:
It was redundant, `pthread_cond_timedwait()` perform this check and will return `EPERM` which is caught in the `VERIFY0(error)` below.